### PR TITLE
CEDS-3221 Split ProcedureCodes page

### PR DESCRIPTION
**Pull request by MaciejRewera**
This PR is only to separate Procedure Code and Additional Procedure Codes
and give each of them a dedicated page.
Therefore logic, validations and implementation are as they were with only
small changes where they were required.

The biggest part of this PR are test suites for controller classes for these
two pages.

### CODE REVIEW

File: app/controllers/declaration/AdditionalProcedureCodesController.scala [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    hc: HeaderCarrier
+  ): Future[Result] =
+    (userInput.additionalProcedureCode, cachedData.additionalProcedureCodes) match {
+      case (_, cachedAdditionalProcedureCodes) if cachedAdditionalProcedureCodes.length >= limitOfCodes =>
+        returnErrorPage(
+          mode,
+          itemId,
+          Seq(("", "declaration.additionalProcedureCodes.maximumAmount.error")),
+          userInput,
+          cachedData.additionalProcedureCodes
```
[the remark this answers is not in the document]
Comment (udinobi) [reply app/controllers/declaration/AdditionalProcedureCodesController.scala:84]: :+1:

File: app/controllers/declaration/ProcedureCodesController.scala [hunk 2]
```diff
@@ -151,103 +92,20 @@ class ProcedureCodesController @Inject()(
...
-          clearDataForProcedureCode(code, itemId, updatedModel)
-        case _ =>
-          updatedModel
-      }
-    }
+      val updatedModel = model.updatedItem(
+        itemId,
+        item => {
+          val newProcedureCodes = item.procedureCodes
+            .fold(ProcedureCodesData(Some(procedureCode.procedureCode), Seq.empty))(_.copy(procedureCode = Some(procedureCode.procedureCode)))
```
Comment (udinobi) [inline app/controllers/declaration/ProcedureCodesController.scala:99]: For a better readibility maybe ?
```
val procedureCode = Some(procedureCode.procedureCode)
...
      .fold(ProcedureCodesData(procedureCode, Seq.empty))(_.copy(procedureCode = procedureCode))
```

File: app/views/declaration/procedureCodes/additional_procedure_codes.scala.html [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+@import forms.declaration.procedurecodes.AdditionalProcedureCodes
+
+@this(
+  govukLayout: gdsMainTemplate,
+  govukTable : GovukTable,
+  addButton: addButton,
+  removeButton: removeButton,
+  errorSummary: errorSummary,
+  sectionHeader: sectionHeader,
+  pageTitle: pageTitle,
```
Comment (udinobi) [inline app/views/declaration/procedureCodes/additional_procedure_codes.scala.html:34]: Not used. Can be removed

File: app/forms/declaration/procedurecodes/AdditionalProcedureCodes.scala [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+import forms.DeclarationPage
+import models.DeclarationType.{CLEARANCE, DeclarationType}
+import models.viewmodels.TariffContentKey
+import play.api.data.Forms.{optional, text}
+import play.api.data.{Form, Forms}
+import play.api.libs.json.Json
+import utils.validators.forms.FieldValidator.{hasSpecificLength, isEmpty, _}
+
+case class AdditionalProcedureCodes(additionalProcedureCode: Option[String])
```
Comment (glidester) [inline app/forms/declaration/procedurecodes/AdditionalProcedureCodes.scala:27]: Should this be pluralised, or just "AdditionalProcedureCode"?

**Comment by udinobi** [comment]
Fine for me. I'll leave Tim to approve it if he also agrees.

**Review by glidester — APPROVED** [review]
:+1:

**Review by udinobi — APPROVED** [review]

**Pull request merged**